src/dbos.ts

```diff
@@ -1325,6 +1326,89 @@ export class DBOS {
     return DBOS.#executor.getEvent(workflowID, key, timeoutSeconds);
   }
 
+  /**
+   * Write a value to a stream.
+   * @param key - The stream key/name within the workflow
+   * @param value - A serializable value to write to the stream
+   */
+  static async writeStream<T>(key: string, value: T): Promise<void> {
+    ensureDBOSIsLaunched('writeStream');
+    if (DBOS.isWithinWorkflow()) {
+      if (DBOS.isInWorkflow()) {
+        const functionID: number = functionIDGetIncrement();
+        return await DBOSExecutor.globalInstance!.systemDatabase.writeStreamFromWorkflow(
+          DBOS.workflowID!,
+          functionID,
+          key,
+          value,
+        );
+      } else if (DBOS.isInStep()) {
+        return await DBOSExecutor.globalInstance!.systemDatabase.writeStreamFromStep(DBOS.workflowID!, key, value);
+      } else {
+        throw new DBOSInvalidWorkflowTransitionError(
+          'Invalid call to `DBOS.writeStream` outside of a workflow or step',
+        );
+      }
+    } else {
+      throw new DBOSInvalidWorkflowTransitionError('Invalid call to `DBOS.writeStream` outside of a workflow or step');
+    }
+  }
+
+  /**
+   * Close a stream by writing a sentinel value.
+   * @param key - The stream key/name within the workflow
+   */
+  static async closeStream(key: string): Promise<void> {
```

Comment:
What if I call `closeStream` and then `writeStream` again? New writes will silently succeed but nothing can be read. Is there a way to warn?